lib/azure/armrest/configuration.rb

```diff
@@ -67,6 +67,12 @@ def self.cache_token(configuration)
       # The environment in which to acquire your token.
       attr_reader :environment
 
+      # The authority URL used to acquire a valid token.
+      attr_accessor :resource
```

Comment:
Is `resource` a too common word (just like service)? Maybe `resource_url`? Same for `authority` and constants if we want to make such renaming.